hack/testing/e2e_deployment.sh

```diff
@@ -0,0 +1,162 @@
+export VERBOSE=1
+
+main() {
+    parse_args "$@"
+    [ "$BOOTSTRAP_OS" ] && bootstrap_os
+    [ "$CLEANUP" ] && trap "oc delete project/$TMP_PROJECT" EXIT
+    [ "$TMP_PROJECT" ] && oc new-project "$TMP_PROJECT"
+    [ "$DEPLOY" ] && setup_for_deployer "$BUILD_IMAGES"
+    [ "$BUILD_IMAGES" ] && build_images
+    run_deployer \
+        "${DEPLOY:+deploy}" \
+        "$([ "$BUILD_IMAGES" -o "$LOCAL_IMAGE" ] && echo 1)"
+    check_deployer
+}
+
+parse_args() {
+    local long tmp
+    long=bootstrap-os,build-images,local-image,deploy,cleanup,tmp-project
+    tmp=$(getopt \
+        --options '' \
+        --long "$long" \
+        --name "$(basename "$0")" -- "$@")
+    eval set -- "$tmp"
+    while [ "$1" != -- ]; do
+        case "$1" in
+            --bootstrap-os) BOOTSTRAP_OS=bootstrap_os; shift;;
+            --build-images) BUILD_IMAGES=build_images; shift;;
+            --local-image) LOCAL_IMAGE=local_image; shift;;
+            --deploy) DEPLOY=deploy; shift;;
+            --cleanup) CLEANUP=cleanup; shift;;
+            --tmp-project) TMP_PROJECT=tmp_project; shift;;
+        esac
+    done
+    if [ ! "$DEPLOY" ]; then
+        [ "$TMP_PROJECT" ] \
+            && args_error 'can only use temporary project when deploying'
+        [ "$CLEANUP" ] \
+            && args_error 'can only use "cleanup" when deploying'
+    fi
+    if [ "$TMP_PROJECT" ]; then
+        TMP_PROJECT=apiman-e2e-test-$( \
+            base64 /dev/urandom | tr -cd [a-z0-9] | head -c 5 || true)
+    else
+        [ "$CLEANUP" ] \
+            && args_error 'can only use "cleanup" with a temporary project'
+    fi
+    [ "$BUILD_IMAGES" -a "$LOCAL_IMAGE" ] \
+        && args_error 'can'\''t use "local image" and "build images" together'
+    return 0
+}
+
+args_error() {
+    printf >&2 "%s: %s\n" "$(basename "$0")" "$@"
+    exit 1
+}
+
+bootstrap_os() {
+    os::util::environment::use_sudo
+    os::util::environment::setup_all_server_vars origin-apiman
+    configure_os_server
+    start_os_server
+    install_registry
+    wait_for_registry
+    export KUBECONFIG="${ADMIN_KUBECONFIG}"
+}
+
+setup_for_deployer() {
+    local build_images=$1 filter
+    os::cmd::expect_success \
+        'oc create -f "$ORIGIN_APIMAN_ROOT/deployer/deployer.yaml"'
+    os::cmd::expect_success \
+        'oc process template/apiman-deployer-account-template | oc create -f -'
+    os::cmd::expect_success \
+        'oadm policy add-role-to-user edit --serviceaccount apiman-deployer'
+    os::cmd::expect_success \
+        "$(echo \
+            oadm policy add-cluster-role-to-user cluster-reader \
+            "system:serviceaccount:$(oc project --short):apiman-gateway")"
+    os::cmd::expect_success \
+        'oc secrets new apiman-deployer nothing=/dev/null'
+    os::cmd::expect_success \
+        'oc label secret/apiman-deployer apiman-infra=deployer'
+    [ "$build_images" ] \
+        && filter="jq '{apiVersion,kind,items:[.items[]|del(.spec.triggers)]}'"
```

Comment:
Can you document any requirements like jq in the README? It exists in the devenv but we need to make sure it's not hard to figure out what you need anywhere else.